hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMPerformanceMetrics.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.ozone.om;
+
+import org.apache.hadoop.metrics2.MetricsSystem;
+import org.apache.hadoop.metrics2.annotation.Metric;
+import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
+import org.apache.hadoop.metrics2.lib.MutableRate;
+
+/**
+ * Including OM performance related metrics.
+ */
+public class OMPerformanceMetrics {
```

Comment:
We need a way to separate end-to-end latency names vs. internal operation names here. It would be helpful to have a prefix for metrics that represent the end-to-end vs. the breakdown of the end-to-end. Example om_request_lookup om_request_lookup_readkey and om_request_lookup_blocktoken. This way, when charting, we can quickly select metrics that break down an end-to-end request. For metrics deeper down the stack and common across multiple requests we can have om_internal_keytable_readkey.

Reply:
Agree. I think in the long run we'd want to introduce a more systematic way to capture metrics with a defined scope, e.g. by adding tags to something queriable. Plus, there should be a utility or common library to handle the repeating metrics logic. E.g the latency capture... I was thinking about doing so, but it needs proper testing/verifying so that capturing metrics doesn't cause an actual impact on the overall perf.